python/lsst/cp/pipe/cpSkyTask.py

```diff
@@ -0,0 +1,172 @@
+        dimensions=("instrument", "calibration_label"),
+    )
+
+    outputExp = cT.Output(
+        name="cpSkyProc",
+        doc="Output combined proposed calibration.",
+        storageClass="ExposureF",
+        dimensions=("instrument", "physical_filter", "detector", "visit"),
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+
+
+class CpSkyTaskConfig(pipeBase.PipelineTaskConfig,
+                      pipelineConnections=CpSkyConnections):
+    maskObjects = pexConfig.ConfigurableField(
+        target=MaskObjectsTask,
+        doc="Object masker to use.",
+    )
+    maskThresh = pexConfig.Field(
+        dtype=float,
+        default=3.0,
+        doc="k-sigma threshold for masking pixels"
+    )
+    mask = pexConfig.ListField(
+        dtype=str,
+        default=["DETECTED", "BAD", "NO_DATA", "SAT"],
+        doc="Mask planes to reject.",
+    )
+
+    largeScaleBackground = pexConfig.ConfigField(
+        dtype=FocalPlaneBackgroundConfig,
+        doc="Large-scale background configuration"
+    )
+    sky = pexConfig.ConfigurableField(
+        target=SkyMeasurementTask,
+        doc="Sky measurement"
+    )
+    background = pexConfig.ConfigField(
+        dtype=BackgroundConfig,
+        doc="Background config.",
+    )
+
+    def setDefaults(self):
+        self.largeScaleBackground.xSize = 256
+        self.largeScaleBackground.ySize = 256
+
+
+class CpSkyTask(pipeBase.PipelineTask,
+                pipeBase.CmdLineTask):
+    ConfigClass = CpSkyTaskConfig
+    _DefaultName = "cpSky"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.makeSubtask("maskObjects")
+        self.makeSubtask("sky")
+
+    def run(self, inputExp, camera):
+        """Preprocess ISR processed exposures for combination to final SKY calibration.
+
+        Parameters
+        ----------
+        inputExp : `lsst.afw.image.Exposure`
+            Pre-processed sky frame data to combine.
+        camera : `lsst.afw.cameraGeom.Camera`
+            CZW confirm this is needed before merge.
+
+        Returns
+        -------
+        outputExp : `lsst.afw.image.Exposure`
+            Sky pre-processed data.
+        """
+        # As constructCalibs SkyTask.processSingleBackground
+        self.maskObjects.run(inputExp, self.config.mask)
+
+        # As constructCalibs SkyTask.measureBackground
+        self.config.largeScaleBackground.xSize = 256
+        self.config.largeScaleBackground.ySize = 256
+        bgModel = FocalPlaneBackground.fromCamera(self.config.largeScaleBackground,
+                                                  camera)
+        bgModel.addCcd(inputExp)
+
+        scale = np.median(bgModel.getStatsImage().getArray())
+
+        # As constructCalibs SkyTask.scatterProcess
+        # Remeasure bkg with scaled version of the FP model removed
+        mi = inputExp.getMaskedImage()
+
+        bg = bgModel.toCcdBackground(inputExp.getDetector(), mi.getBBox())
```

Comment:
I think this has to be missing the crucial stage of gathering the `bgModel`s from all the CCDs in the visit. I very much doubt that this works as it's intended.

Reply:
I would rather not hold up the rest of the ticket for the sky calibration (which I've never been completely sure about anyway).  Is there documentation on the algorithm that I can link the ticket covering this issue (DM-22534).

Reply:
If the sky construction isn't to be part of this ticket, then you should put that in the ticket description, and remove this code completely. The algorithm is described fairly broadly [here](https://community.lsst.org/t/sky-subtraction/2415).

The flat construction has the same problem.